lib/galaxy/config.py

```diff
@@ -363,9 +365,9 @@ def __init__( self, **kwargs ):
             # so dependending on the context a better default can be used (
             # request url in a web thread, Docker parent in IE stuff, etc...)
             galaxy_infrastructure_url = "http://localhost"
-            port = self.guess_galaxy_port()
-            if port:
-                galaxy_infrastructure_url += ":%s" % (port)
+            web_port = self.galaxy_infrastructure_web_port or self.guess_galaxy_port()
+            if web_port:
+                galaxy_infrastructure_url += ":%s" % (web_port)
```

Comment:
> I don't know the IP address of the VM ahead of time.

OH, now this is starting to make more sense. So IPython is accessing the wrong port (for galaxy in the docker host). You're setting it explicitly because `guess_galaxy_port()` is giving you a wrong port # number also?